src/main/scala/com/raphtory/core/analysis/GraphLens.scala

```diff
@@ -48,29 +49,32 @@ final case class GraphLens(
     viewTimer.update(System.currentTimeMillis() - startTime)
     result
   }
-
+  def getVertices(): ParIterable[Vertex] = {
+    vertexCount.set(vertexMap.size)
+    vertexMap.map(x=>x._2)
+  }
   def getMessagedVertices(): ParIterable[Vertex] = {
     val startTime = System.currentTimeMillis()
-    val result    = getVertices.filter(_.hasMessage())
-    messageFilter = true
+    val result    = vertexMap.collect {
+      case (k, vertex) if vertex.hasMessage() => vertex
+    }
     viewTimer.update(System.currentTimeMillis() - startTime)
+    vertexCount.set(result.size)
     result
   }
 
-  def getVertices(): ParIterable[Vertex] = vertexMap.map(x=>x._2)
-
-  def checkVotes(): Boolean =
-    if (messageFilter)
-      getMessagedVertices().size == voteCount
-    else
-      vertexMap.size == voteCount
+  def checkVotes(): Boolean = vertexCount.get() == voteCount.get()
 
   //TODO hide away
   def sendMessage(msg: VertexMessage): Unit = messageHandler.sendMessage(msg)
 
 
-  def vertexVoted(): Unit = voteCount += 1
-  def nextStep(): Unit    = superStep += 1
+  def vertexVoted(): Unit = voteCount.incrementAndGet()
+  def nextStep(): Unit    = {
+    voteCount.set(0)
+    vertexCount.set(0)
```

Comment:
this does not make sense. vertex count should be = static all the time

Reply:
If the user is calling .getMessagedVertices() this number will change each superstep so needs to be reset.